src/backend/vulkan/src/device.rs

```diff
@@ -471,17 +472,38 @@ impl d::Device<B> for Device {
                 },
             });
 
-            info_multisample_states.push(vk::PipelineMultisampleStateCreateInfo {
+            // Multisampling
+            //
+            // Default multi sampling state, required even if we only have a single sample attachment.
+            let mut multisampling_state = vk::PipelineMultisampleStateCreateInfo {
                 s_type: vk::StructureType::PipelineMultisampleStateCreateInfo,
                 p_next: ptr::null(),
                 flags: vk::PipelineMultisampleStateCreateFlags::empty(),
-                rasterization_samples: vk::SAMPLE_COUNT_1_BIT, // TODO
-                sample_shading_enable: vk::VK_FALSE, // TODO
-                min_sample_shading: 0.0,  // TODO
-                p_sample_mask: ptr::null(), // TODO
-                alpha_to_coverage_enable: vk::VK_FALSE, // TODO
-                alpha_to_one_enable: vk::VK_FALSE, // TODO
-            });
+                rasterization_samples: vk::SAMPLE_COUNT_1_BIT,
+                sample_shading_enable: vk::VK_FALSE,
+                min_sample_shading: 0.0,
+                p_sample_mask: ptr::null(),
+                alpha_to_coverage_enable: vk::VK_FALSE,
+                alpha_to_one_enable: vk::VK_FALSE,
+            };
+
+            if let Some(ref multisample) = desc.multisampling {
```

Comment:
can we please do `let multisampling = match desc.multisampling {...}` instead of mutating it?